File: frontend/src/types/credentials.ts
type DateTimeString = string;

export interface CredentialCreate {
    name: string;
    description: string;
    payload: { [key: string]: string };
}

export interface Credential {
    id: string;
    issuer_id: number;
    name: string;
    description: string;
    payload: { [key: string]: string };
    created_date: DateTimeString;
    signature: string;
}

File: backend/app/models/credentials.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime 

class CredentialsCreate(BaseModel):
    name: str
    description: str
    payload: dict

class Credentials(BaseModel):
    id: UUID
    issuer_id: int
    name: str
    description: str
    payload: dict
    signature: str
    created_date: datetime